staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/validation/validation.go

```diff
@@ -287,7 +334,7 @@ func ConvertJSONSchemaPropsWithPostProcess(in *apiextensions.JSONSchemaProps, ou
 		out.VendorExtensible.AddExtension("x-kubernetes-embedded-resource", true)
 	}
 	if len(in.XListMapKeys) != 0 {
-		out.VendorExtensible.AddExtension("x-kubernetes-list-map-keys", in.XListMapKeys)
+		out.VendorExtensible.AddExtension("x-kubernetes-list-map-keys", convertSliceToInterfaceSlice(in.XListMapKeys))
```

Comment:
was this an existing bug? trying to understand what this is doing/fixing

Reply:
Yes, pre-existing. The conversion function placed `[]string` types in this list which expects `[]interface{}`. When attempting to use `VendorExtensible` library functions with the result of this to fetch extensions, `x-kubernetes-list-map-keys` was treated as not present. 

More discussion here: https://github.com/kubernetes/kube-openapi/pull/413#issuecomment-1636119257

Reply:
ok... is that a fix we should backport independent of this feature?

Reply:
I'm not sure if prior releases would have exercised this bug, but I will open a cherry pick as far back as I can go.